torch/csrc/jit/pickler.cpp

```diff
@@ -218,18 +219,23 @@ void Pickler::pushIValue(const IValue& ivalue) {
     }
   }
   pushIValueImpl(ivalue);
-  if (ivalue.isPtrType()) {
+  if (ivalue.isPtrType() && !ivalue.isString() && !ivalue.isTuple()) {
```

Comment:
this exact check is also repeated below. Maybe, we should consider moving it into a helper  (`isEfficientlyMemoizablePtrType`) or moving the whole `isPtrType` case into its own function and just calling `pushIValueImpl`, otherwise. Even though it looks like the second check is what that gives us size savings, refactoring it into the helper might help us avoid miscopying it in the future